carbon-route.html

```diff
@@ -223,16 +223,16 @@
     __resetProperties: function() {
       this._setActive(false);
       this._matched = null;
-      //this.tail = { path: null, prefix: null, queryParams: null };
-      //this.data = {};
+      this.tail = { path: null, prefix: null, __queryParams: null };
+      this.data = {};
```

Comment:
This does affect me too, I have a route `/:page` then a sub-route of `/foo/:id`.

If you navigate to `/foo/1234`, then to `/foo`, the `id` property will still be set to `1234`.

Workaround for this is probably to check _both_ `active` and `data.id`.